src/bin/hal/cmd/bech32/decode.rs

```diff
@@ -0,0 +1,42 @@
+use bech32lib::{decode, FromBase32};
+use clap;
+
+use cmd;
+use hal;
+
+pub fn subcommand<'a>() -> clap::App<'a, 'a> {
+	cmd::subcommand("decode", "decode bech32 format").args(&[
+		cmd::opt_yaml(),
+		cmd::opt(
+			"convert-bits",
+			"Attempt to convert payload from 5-bit to 8-bit values.\nNOTE: Does not work for BIP-173 addresses."
+		)
+		.short("c")
+		.required(false),
+		cmd::arg("string", "a bech32 string").takes_value(true).required(true),
+	])
+}
+
+pub fn execute<'a>(matches: &clap::ArgMatches<'a>) {
+	let s = matches.value_of("string").expect("missing required argument");
+	let result = decode(&s);
+	if result.is_err() {
+		panic!("Decode failure: {:?}", result.unwrap_err());
```

Comment:
Why did you decide to do this instead of an `.expect("bech32 decode failure")`?

Reply:
Is there a way to pass error details to `expect`?

Reply:
Ok, it looks like it does output details. Will update.

Reply:
Yeah expect always prints the error it got into the panic message. If you want to provide any extra information, I'd do `.expect(format!(...`.